### DISCUSSION TITLE
how to run test-backend-ops in single threaded mode

### DISCUSSION THREAD

**Discussion by amritahs** [ggml-org/llama.cpp#9920] [Q&A] [answered]
I am trying to debug llamafile_sgemm API and I am invoking just MUL_MAT operations on CPU backend. I would like to know if it is possible to run test-backend-ops in single threaded mode so that debugging becomes easier

**Comment by slaren** [comment]
You would have to modify the call to `ggml_backend_set_n_threads_fn` in `test-backend-ops.cpp` to set the number of threads to 1 instead of `std::thread::hardware_concurrency()`.

**Comment by nikhil-arm** [comment, marked as answer]
You can set this to 1 and recompile as well 
https://github.com/ggerganov/llama.cpp/blob/afd9909a6481402844aecefa8a8908afdd7f52f1/ggml/include/ggml.h#L229
